### DISCUSSION TITLE
horizontalLayout on lineChart, but not columnChart

### DISCUSSION THREAD

**Discussion by Curwe** [patrykandpatrick/vico#579] [Ideas]
v. 1.4.0-beta01

I'm running both a lineChart + columnChart in my project. 
While looking at the sample project (chart 3) you applied `horizontalLayout = HorizontalLayout.fullWidth()` to render the lineChart all the way to the edge of the chart. 

Is it possible to have this behavior done without it affecting the columnChart? Or somehow make the lineChart "item" stretch over the entire x-axis step not only being a dot in the center? 

I've attached a screenshot of that I want it to look like. But today the line is rendered in the center behind the column item. 
 
![Screenshot 2024-02-20 at 08 11 23](https://github.com/patrykandpatrick/vico/assets/135611838/18a53337-ee50-48c4-930a-6295a4036e93)

**Comment by patrickmichalik** [comment]
Hello! `HorizontalLayout.FullWidth` applies to all `Chart`s in a `ComposedChart` because the _x_ scales have to align. You can shift the line by updating its _x_ values accordingly and using `AxisValueOverrider` to set the minimum _x_ value to the _x_ value of the first column.

You can combine this with a step `LineSpec.PointConnector` to replicate a step chart, but if your chart has a `Marker`, the result might be suboptimal, as the points will be at the edges of the steps, and an extra point will be needed at the end for the line to stretch to the end of the chart. We’ll be looking into adding step charts to Vico—I’m moving this discussion to the “Ideas” category.
